internal/os/filesystem/api.go

```diff
@@ -48,31 +61,34 @@ func pathValid(path string) (bool, error) {
 //   https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/test-path?view=powershell-7
 // for a remote path, determines whether connection is ok
 //   e.g. in a SMB server connection, if password is changed, connection will be lost, this func will return false
-func (APIImplementor) PathValid(path string) (bool, error) {
+func (filesystemAPI) PathValid(path string) (bool, error) {
 	return pathValid(path)
 }
 
-func (APIImplementor) Mkdir(path string) error {
+// Mkdir makes a dir with `os.MkdirAll`.
+func (filesystemAPI) Mkdir(path string) error {
 	return os.MkdirAll(path, 0755)
 }
 
-func (APIImplementor) Rmdir(path string, force bool) error {
+// Rmdir removes a dir with `os.Remove`, if force is true then `os.RemoveAll` is used instead.
+func (filesystemAPI) Rmdir(path string, force bool) error {
 	if force {
 		return os.RemoveAll(path)
 	}
 	return os.Remove(path)
 }
 
-func (APIImplementor) LinkPath(oldname, newname string) error {
+// CreateSymlink creates newname as a symbolic link to oldname.
+func (filesystemAPI) CreateSymlink(oldname, newname string) error {
 	return os.Symlink(oldname, newname)
 }
 
-// IsMountPoint - returns true if its a mount point.
+// IsSymlink - returns true if tgt is a mount point.
 // A path is considered a mount point if:
 //  - directory exists and
 //  - it is a soft link and
 //  - the target path of the link exists.
-func (APIImplementor) IsMountPoint(tgt string) (bool, error) {
+func (filesystemAPI) IsSymlink(tgt string) (bool, error) {
```

Comment:
need more comments on what is returned

if tgt path does not exist, it return error
if tgt path exists, but the source path tgt points to does not exist, it return false without error

Reply:
done

Reply:
sorry, maybe add similar comments into API definition too?